test/val/val_decoration_test.cpp

```diff
@@ -646,119 +646,6 @@ TEST_F(ValidateDecorations, RuntimeArrayOfDescriptorSetsIsAllowed) {
   EXPECT_EQ(SPV_SUCCESS, ValidateAndRetrieveValidationState());
 }
 
-// #version 440
-// #extension GL_EXT_nonuniform_qualifier : enable
-// layout(binding = 1) uniform sampler2D s2d[][2];
-// layout(location = 0) in nonuniformEXT int i;
-// void main()
-// {
-//     vec4 v = texture(s2d[i][i], vec2(0.3));
-// }
-TEST_F(ValidateDecorations, RuntimeArrayOfArraysOfDescriptorSetsIsDisallowed) {
```

Comment:
Can you add some equivalents of these tests in val_memory_test.cpp that get caught be your check?